src/exberry_int/integration_exberry.py

```diff
@@ -102,17 +104,25 @@ async def request_market_data():
 
 
     async def producer_coro(ws):
+        request_to_send = None
         try:
             while True:
                 await session_started.wait()
                 LOG.info("Awaiting next message to send...")
-                request_to_send = await outbound_queue.get()
+                if pending_outbound_queue.qsize() > 0:
```

Comment:
I think the priority queue makes this irrelevant, but there's also `get_nowait` - which is a non blocking `get` that throws an exception on an empty queue. This lets you check queue size and dequeue in a single statement, thus avoiding the race condition you have here.